python/lib.c

```diff
@@ -231,13 +236,27 @@ static PyObject* impl_metric(simsimd_metric_kind_t metric_kind, PyObject* const*
         npy_intp dims[1] = {count_max};
         PyArray_Descr* descr = PyArray_DescrFromType(NPY_FLOAT32);
         PyArrayObject* output_array = (PyArrayObject*)PyArray_NewFromDescr( //
-            &PyArray_Type, descr, 1, dims, NULL, distances, NPY_ARRAY_OWNDATA | NPY_ARRAY_C_CONTIGUOUS, NULL);
+            &PyArray_Type, descr, 1, dims, NULL, distances, NPY_ARRAY_WRITEABLE, NULL);
 
         if (!output_array) {
             free(distances);
             goto cleanup;
         }
 
+        PyObject *wrapper = PyCapsule_New(distances, "wrapper", (PyCapsule_Destructor)&free_capsule);
```

Comment:
I am not sure, that there is way to avoid creating a capsule, but we probably can add smth like `out` in the signature, so it will be:

```
query: numpy.ndarray = ...
embeddings: numpy.ndarray = ...

scores: numpy.ndarray = numpy.empty(
    shape=(query.shape[0], embeddings.shape[0]), 
    dtype=numpy.dtype("float32")
)

simsimd.cdist(embeddings, query, out=scores)
```

In this way, you can create an output array only once, and then just reuse it.

Reply:
That's a good idea! Can you implement it? We also must document that behavior, as it [goes beyond SciPy functionality](https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.distance.cdist.html).

Reply:
Okay, I will take a look at it, later this week